### PR TITLE
Add `identity` function to `scala.Function`.

### PR DESCRIPTION
**Pull request by He-Pin**
Now, Scala has what's Java has too.

### CODE REVIEW

**Comment by joshlemer** [comment]
Why is there need for `Function.identity` when we have already `Predef.identity`?

**Comment by He-Pin** [comment]
> Why is there a need for `Function.identity` when we have already `Predef.identity`?

For someone comes from Java will try to find the `identity` method on `Function`, and `Predef#identity` is a method be to convert to Function, not itself a Function1.

**Comment by lrytz** [comment]
I don't think we should duplicate functions like this. There's little value in moving definitions around like this. Maybe one day we'll do a Predef cleanup, but for 2.13 we're too close to the RC.

File: src/library/scala/Function.scala [hunk 1]
```diff
@@ -28,6 +28,15 @@ object Function {
...
   def const[T, U](x: T)(y: U): T = x
 
+  /**
+   * Returns a function that always returns its input argument as result.
+   * Which could be used as a function argument passed to `map` or `flatMap` methods.
+   * Learn more from [[https://en.wikipedia.org/wiki/Identity_function Identity_function]]
+   * @tparam T the type of the input and output value of the function
+   * @return a function that always returns its input argument as result.
+   */
+  def identity[T]: T => T = t => t
```
Comment (viktorklang) [inline src/library/scala/Function.scala:38]: As is, this will allocate a new instance for each invocation, IMO this should return a cached instance.

Comment (He-Pin) [reply src/library/scala/Function.scala:38]: it was constIdentity.asInstanceOf[T=>T] ,but I  changed that before pr.

This thing is what I learnt from you:stuck_out_tongue_winking_eye:

Comment (He-Pin) [reply src/library/scala/Function.scala:38]: @viktorklang like this:

https://github.com/hepin1989/scala/commit/c29ac874d18dd1f8f81e81cb74aab8b8ac1680a9#diff-cdead8ee05a4284557bc258dc86e300dR39

Comment (viktorklang) [reply src/library/scala/Function.scala:38]: :-)))

**Pull request closed without merging**